src/modules/teams.js

```diff
@@ -0,0 +1,200 @@
+import { API_BASE_URL } from '../constants/global'
+import { unique } from '../helpers/utils';
+import { toastr } from 'react-redux-toastr';
+
+// Actions
+const GET_TEAM = 'teams/GET_TEAM';
+const SAVE_TEAM = 'teams/SAVE_TEAM';
+const SET_EDITING = 'teams/SET_EDITING';
+const SET_LOADING = 'teams/SET_LOADING';
+const SEND_NOTIFICATIONS = 'teams/SEND_NOTIFICATIONS'
+
+// Reducer
+const initialState = {
+  data: null,
+  sendNotifications: false,
+  editing: false,
+  loading: true
+};
+
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    case GET_TEAM: {
+      const team = action.payload;
+      if (team) return Object.assign({}, state, { data: team });
+      return state;
+    }
+    case SAVE_TEAM: {
+      const team = action.payload.team;
+      return {
+        ...state,
+        teams: { ...state.teams, ...team }
+      };
+    }
+    case SET_EDITING:{
+      return Object.assign({}, state, { editing: action.payload });
+    }
+    case SET_LOADING:{
+      return Object.assign({}, state, { loading: action.payload });
+    }
+    case SEND_NOTIFICATIONS:{
+      return Object.assign({}, state, { sendNotifications: action.payload });
+    }
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+
+export function getTeam(userId) {
+  const url = `${API_BASE_URL}/teams/user/${userId}`;
+  return (dispatch, state) => {
+    return fetch(url, {
+      headers: {
+        Authorization: `Bearer ${state().user.token}`
+      }
+    })
+      .then((response) => {
+        if (response.ok) return response.json();
+        throw Error(response.statusText);
+      })
+      .then((response) => {
+        const team = response.data;
+        dispatch({
+          type: GET_TEAM,
+          payload: team
+        });
+        dispatch({
+          type: SET_LOADING,
+          payload: false
+        });
+        return team;
+      })
+      .catch((error) => {
+        console.warn('error', error)
```

Comment:
Dont forget to set loading to false here also. Otherwise we will be spinning forever.